internal/rcm/api.go

```diff
@@ -204,8 +204,19 @@ func (api *API) submit(writer http.ResponseWriter, request *http.Request, _ http
 		return
 	}
 
+	size := imageType.Size(0)
+	manifest, err := imageType.Manifest(nil, repoConfigs, packages, buildPackages, size)
+	if err != nil {
+		writer.WriteHeader(http.StatusBadRequest)
+		_, err := writer.Write([]byte(err.Error()))
+		if err != nil {
+			panic("Failed to write response")
+		}
+		return
+	}
+
 	// Push the requested compose to the store
-	composeID, err := api.store.PushCompose(imageType, &blueprint.Blueprint{}, repoConfigs, packages, buildPackages, 0, nil)
+	composeID, err := api.store.PushCompose(manifest, imageType.Name(), &blueprint.Blueprint{}, 0, nil)
```

Comment:
Shouldn't `size` be passed here?

Reply:
And also `imageType` instead of `imageType.Name()`?

Reply:
I think you're looking at a commit in the middle of the PR. This is both addressed in the final one.

Reply:
Ah, makes sense. Thanks!